lib/rspec-puppet/support.rb

```diff
@@ -254,6 +254,19 @@ def build_catalog_without_cache(nodename, facts_val, trusted_facts_val, hiera_co
     end
 
     def stub_facts!(facts)
+      if facts['operatingsystem'] && facts['operatingsystem'].to_s.downcase == 'windows'
+        # If we're not running in windows but are testing a catalogue destined
+        # for a windows host, pretend to be windows in the least awful way that
+        # I can think of.
+        unless Puppet::Util::Platform.windows?
+          Puppet::Util::Platform.pretend_windows
+          # On windows, puppet validates the value of the autosign setting as
+          # an absolute path unless it's disabled.
+          Puppet.settings[:autosign] = false
```

Comment:
Is there a risk of breaking spec tests; if someone happened to be asserting this was true? I think if so; we could put somewhere in docs that we have this restriction when testing a module with Windows facts